build-tools/automation/yaml-templates/setup-jdk-variables.yaml

```diff
@@ -0,0 +1,26 @@
+parameters:
+  jdkMajorVersion: $(DefaultJavaSdkMajorVersion)
+  useAgentJdkPath: true
+
+steps:
+- pwsh: |
+    $agentOS="$(Agent.OS)"
+    $agentArch="$(Agent.OSArchitecture)"
```

Comment:
There is likely a bug here because `$(Agent.OSArchitecture)` will be `ARM64`, the environment variable is `JAVA_HOME_17_arm64`, and environment variable names are case sensitive on Mac/Linux.  But I can fix this in my ARM64 PR since I can test it easier.